consensus/hotstuff/verifier.go

```diff
@@ -10,8 +10,8 @@ import (
 type Verifier interface {
```

Comment:
Would suggest to be very clear here about the performance tradeoff we are making. Suggested goDoc for `Verifier` interface:

```golang
// Verifier is the component responsible for the cryptographic integrity of
// votes, proposals and QC's against w.r.t. the block they are signing.
// Overall, there are two criteria for the validity of a vote and QC:
//  (1) the signer ID(s) must correspond to authorized consensus participants
//  (2) the signature must be cryptographically valid.
// Note that Verifier only implements (2). This API design allows to decouple
//  (i) the common logic for checking that a super-majority of the consensus
//      committee voted
// (ii) the handling of combined staking+RandomBeacon votes (consensus nodes)
//      vs only staking votes (collector nodes)
// On the one hand, this API design makes code less concise, as the two checks
// are now distributed over API boundaries. On the other hand, we can avoid
// repeated Identity lookups in the implementation, which increases performance.
type Verifier interface {
   . . .
}
```

Reply:
Thanks for clarifying the doc 🙇